azure-pipelines.yml

```diff
@@ -0,0 +1,98 @@
+        variables:
+          sdkVersion: 3.1.x
+          buildConfiguration: 'Release'
+      # build using dotnet 5 
+        steps:
+        - task: UseDotNet@2
+          displayName: dotnet sdk 2.x
+          inputs:
+            packageType: 'sdk'
+            version: '5.x'
+            installationPath: $(Agent.ToolsDirectory)/dotnet
+            workingDirectory: 'FakebookNotifications/FakebookNotifications.WebApi/'
+        # run build
+        - script: dotnet build --configuration $(buildConfiguration)
+          displayName: dotnet build
+          workingDirectory: 'FakebookNotifications/FakebookNotifications.WebApi/'
+        # publish build to artifact
+        - script: dotnet publish --configuration $(buildConfiguration) --output publish
+          displayName: dotnet publish
+          workingDirectory: 'FakebookNotifications/FakebookNotifications.WebApi/'
+        
+        - publish: 'FakebookNotifications/FakebookNotifications.WebApi/publish'
+          artifact: app
+  - stage: SonarCloud
+    jobs:
+    - job: build
+      # use ubuntu latest
+      pool:
+        vmImage: ubuntu-latest
+        # use an earlier version of .net because sonar cloud doesn't like .net 5
+      variables:
+        sdkVersion: 3.1.x
+        buildConfiguration: 'Release'
+        #use sonar cloud token 
+      steps:
+      - task: UseDotNet@2
+        displayName: dotnet sdk 2.x
+        inputs:
+          packageType: 'sdk'
+          version: '2.x'
+      - task: SonarCloudPrepare@1
+        inputs:
+          SonarCloud: 'SonarCloud Token'
+          organization: '2011-fakebook-project3'
+          scannerMode: 'MSBuild'
+          projectKey: '2011-fakebook-project3_notifications'
+          extraProperties: |
+                sonar.exclusions=**/lib/**
+                sonar.cs.opencover.reportsPaths = $(Agent.TempDirectory)/**/coverage.opencover.xml
+    # use the latest .net
+      - task: UseDotNet@2
+        displayName: 'Use .NET 5 SDK'
+        inputs:
+          packageType: 'sdk'
+          version: '5.x'
+          installationPath: $(Agent.ToolsDirectory)/dotnet
+          workingDirectory: 'FakebookNotifications/FakebookNotifications.WebApi/'
+      # run dotnet tests
+      - task: DotNetCoreCLI@2
+        displayName: dotnet test
+        inputs:
+          command: 'test'
+          arguments: '--configuration $(BuildConfiguration) --collect:"XPlat Code Coverage" -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Format=opencover'
+          publishTestResults: true
+          workingDirectory: 'FakebookNotifications/'
+        # publish the dotnet tests
+      - script: dotnet publish --configuration $(buildConfiguration) --output publish
+        displayName: dotnet publish
+        workingDirectory: 'FakebookNotifications'
+      # analyze the tests
+      - task: SonarCloudAnalyze@1
+        displayName: sonar run analysis
+      # publish to sonar cloud
```

Comment:
this step does not publish to sonar cloud. the run analysis step does that. this step publishes the quality gate result to the build in azure devops, so that we can click on the build in azure devops and see if the quality gate passed or failed without going to sonarcloud